### PR TITLE
feat(wmg): Newsletter Signup

### PR DESCRIPTION
**Pull request by ashin-czi**
## Reason for Change

- #4453 

Readability: @prathapsridharan 
Functionality: @atarashansky 

## Changes

- add
    - Adding new Newsletter component
- remove
- modify

## Testing steps

rdev: https://newsletter-signup-frontend.rdev.single-cell.czi.technology/gene-expression
newsletter direct link: https://newsletter-signup-frontend.rdev.single-cell.czi.technology/gene-expression?newsletter_signup=true

1. Navigate to gene expression, datasets, or collections page
2. Click "subscribe" button in the banner to open the modal and ensure modal opens with content and logo
3. Enter an invalid and/or valid email address in the email input box
4. Press "subscribe" button
5. If an invalid email was entered, you should see an error message underneath the input box
6. If a valid email was entered, ensure you see "Thanks for subscribing!" text after submission
7. Click the X button to close out the modal
7. Refresh the page to reset the newsletter banner to input an email again

Home landing page
1. Scroll to the bottom of the page
2. Ensure banner is attached to the footer of the page (note there is no modal to open on the home page)
3. Enter an invalid/valid email address
4. Press "submit" button
5. If an invalid email was entered, you should see an error message underneath the input box
6. If a valid email was entered, you should see "Thanks for subscribing!" text after submission
7. Refresh the page to reset the newsletter banner to input an email again

## Notes for Reviewer

Implementation is almost exactly how Napari Hub integrates with Hubspot in order to prevent saving cookies. Their PR for integrating with Hubspot can be viewed here: https://github.com/chanzuckerberg/napari-hub/pull/661/files

### CODE REVIEW

File: frontend/tests/features/wheresMyGene/wheresMyGene.test.ts [hunk 1]
```diff
@@ -791,6 +791,93 @@ describe("Where's My Gene", () => {
...
+      expect(modalContent.getByText("Unsubscribe at any time.")).toBeTruthy();
+
+      // Close modal
+      await getButtonAndClick(page, NEWSLETTER_MODAL_CLOSE_BUTTON);
+
+      await waitForElementToBeRemoved(page, NEWSLETTER_MODAL_CONTENT);
+    });
+
+    test("Newsletter Modal - Validate Email", async ({ page }) => {
+      await goToPage(`${TEST_URL}${ROUTES.WHERE_IS_MY_GENE}`, page);
```
Comment (prathapsridharan) [inline frontend/tests/features/wheresMyGene/wheresMyGene.test.ts:834]: Is this missing a couple of tests?

1. valid email format. ex: `dbrooks@grizzlies.com`
2. valid email format but email does not exist. Maybe you get one of these SMTP errors in the backend that shows up something else on the frontend? https://support.google.com/a/answer/3221692?hl=en
3. Lastly a valid email address that does exist.

**Review by prathapsridharan — COMMENTED** [review]
@ashin-czi - 
1. I mostly focused on the tests. I posted a couple of questions. 

2. I went through the frontend code rather quickly because I am not very qualified there but I did grasp enough to get the gist of it. Looks fine to me and the rdev is a demonstration of its functionality. I will leave it to someone more experienced in frontend to give pointed feedback on the code.

**Review by prathapsridharan — APPROVED** [review]
LGTM

**Review by atarashansky — APPROVED** [review]
LGTM! Great job as always

**Pull request merged**